src/gc/gcpriv.h

```diff
@@ -781,6 +779,9 @@ class generation
     int             gen_num;
 
 #ifdef FREE_USAGE_STATS
+    #pragma message ("FREE_USAGE_STATS changes the layout of the 'generation' class, which is known to the DAC. " \
```

Comment:
We should prefer design choices that allow you to make changes in the GC without changing its contract. I would go via the size in dacvar.

>  implementation complexity.

This should be pretty minor - one or two macros or helper methods should hide the detail.